docs/development/design/build-images.rst

```diff
@@ -191,94 +299,122 @@ Examples of these versions are:
 
-How do we remove an old Python version?
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If it's strongly needed and we can't wait for a new base image,
+we could install it at build time in a similar way as we do with ``build.apt_packages`` as a temporal workaround.
 
-At some point an old version of Python will be deprecated (eg. 3.4) and will be removed.
-To achieve this, we can just remove the Docker image affected: ``ubuntu20-py34``,
-once there are no users depending on it anymore.
+How do we create a mirror for each language?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A mirror can be created with ``wget`` together with ``rclone``:
+
+#. Download all the files from the official mirror:
+
+   .. code:: bash
+
+      # https://stackoverflow.com/questions/29802579/create-private-mirror-of-http-nodejs-org-dist
+      wget --mirror --convert-links --adjust-extension --page-requisites --no-parent -e robots=off http://nodejs.org/dist
+
+#. Upload all the files to S3:
+
+   .. code:: bash
+
+      # https://rclone.org/s3/
+      rclone sync -i nodejs.org s3:languages
+
+
+.. note::
+
+   Downloading a copy of the official mirror took 15m and 52Gb.
+
+How local development will work with the new approach?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Local development will require scripts to clone the official mirrors for each language and upload them to MinIO (S3).
+Besides, a script to define a set of Python version, pre-compile them and also upload them to S3.
 
-We will know which projects are using these images because they are pinning it in the config file.
-We could show a message in the build output page and also send them an email with the EOL date for this image.
 
 Deprecation plan
 ----------------
 
-It seems we have ~50Gb free on builders disks.
-Considering that the new images will be sized approximately (built locally as test):
-
-* ``ubuntu20-base``: ~5Gb
-* ``ubuntu20-py27``: ~150Mb
-* ``ubuntu20-py36``: ~210Mb
-* ``ubuntu20-py39``: ~20Mb
-* ``ubuntu20-conda47``: ~713Mb
+After this design document gets implemented and tested,
+all our current images (``stable``, ``latest``, ``testing``) will be deprecated and their usage will be de-motivated.
+However, we could keep them on our builders to give users a good time to migrate their projects to the new ones.
+New projects shouldn't be able to select these images and they will be forced to use ``build.os`` and ``build.languages``.
 
-which is about ~6Gb in total, we still have plenty of space.
+We may want to keep only the latest Ubuntu LTS releases available in production,
+with a special consideration for our current Ubuntu 18.04 LTS on ``stable``, ``latest`` and ``testing`` because 100% of the projects depend on them currently.
+Once Ubuntu 22.04 LTS is released, we should communicate that Ubuntu 20.04 LTS is deprecated,
+and give users 2 years to migrate to a newer image (1 year less than the official Ubuntu support for LTS images)
```

Comment:
Why not just use the actual support window for the release? This seems arbitrary.